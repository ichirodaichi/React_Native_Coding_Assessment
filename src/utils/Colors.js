export const COLORS = {
    Black: 'black',

    White: '#FFFFFF',

    ThemeColor: '#2F66F6',

    LightBlue: '#D7D9E4',

    Grey: '#696F8C',
    bgColor: '#F8F8FF',
    tabBg:'rgba(118, 118, 128, 0.12)'

}