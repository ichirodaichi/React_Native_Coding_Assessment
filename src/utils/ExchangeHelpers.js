const USDtoBitcoinRate = 0.000050;
const USDtoETHRate = 0.00063;

function ConvertUSDToBitcoin(uSDValue) {
    return uSDValue * USDtoBitcoinRate;
}

function ConvertUSDToETH(uSDValue) {
    return uSDValue * USDtoETHRate;
}

module.exports = { ConvertUSDToBitcoin, ConvertUSDToETH }