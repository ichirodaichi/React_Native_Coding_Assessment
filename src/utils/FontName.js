export const FONT_NAME={
    Brown:'Brown-Regular',
    Brown_Bold:'Brown-Bold'
  
}