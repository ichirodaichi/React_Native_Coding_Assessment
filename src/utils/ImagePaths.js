const IMAGE_PATHS = {
    BackIcon: require('../assets/icons/back/Back.png'),
    InfoIcon: require('../assets/icons/info/Info_light.png'),
    ArrowIcon: require('../assets/icons/arrow/icon.png'),
    BitcoinIcon: require('../assets/icons/bitcoin/Bitcoin.png'),
    EthereumIcon: require('../assets/icons/ethereum/Ethereum.png'),
    ExchangeIcon: require('../assets/icons/exchangeIcon/icon.png'),
    //

    Illustration: require('../assets/images/illustration/Illustration.png')


}
export default IMAGE_PATHS;