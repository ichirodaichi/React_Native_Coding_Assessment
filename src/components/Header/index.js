import React from 'react';
import {
  View,
  Text,
  Image,
  TouchableOpacity,
} from 'react-native';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from 'react-native-responsive-screen';
import { COLORS } from '../../utils/Colors';
import { FONT_SIZE } from '../../utils/FontSizes';
import IMAGE_PATHS from '../../utils/ImagePaths';

const Header = props => {
  const { headerText, onPressBack, onPressRightIcon } =
    props;
  return (
    <View style={{ paddingBottom: wp(0.4) }}>
      <View
        style={{
        
          position: 'relative',
          paddingHorizontal: wp(2),
          alignItems: 'center',
          justifyContent: 'center',
          alignSelf: 'center',
          width: wp(100),
          flexDirection: 'row',
          backgroundColor: COLORS.bgColor,
          height: wp(100) / 7,
        }}>


        <View
          style={{
            width: wp(92),

            justifyContent: 'space-between',
            flexDirection: 'row',
          }}>
          <TouchableOpacity style={{ justifyContent: 'center', alignItems: 'center' }} onPress={onPressBack}>
            <Image source={IMAGE_PATHS.BackIcon} resizeMode={'contain'} />

          </TouchableOpacity>
          <Text
            style={{
              fontSize: FONT_SIZE.Medium,
              fontWeight: 'bold'
            }}>
            {headerText}
          </Text>

          <TouchableOpacity>
            <Image source={IMAGE_PATHS.InfoIcon} resizeMode={'contain'} />
          </TouchableOpacity>
        </View>
      </View>
    </View>
  );
};

export default Header
