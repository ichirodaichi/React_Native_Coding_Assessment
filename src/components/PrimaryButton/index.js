import React from 'react';
import {
  SafeAreaView,
  Text,
  View,
  StyleSheet,
  Dimensions,
  TouchableOpacity,
} from 'react-native';
import { widthPercentageToDP } from 'react-native-responsive-screen';
import { COLORS } from '../../utils/Colors';
import { FONT_SIZE } from '../../utils/FontSizes'

const { width } = Dimensions.get('window');
const { height } = Dimensions.get('window');

const PrimaryButton = ({ text, onPress, disabled }) => {
  return (
    <TouchableOpacity
      style={
        disabled ? styles.disabledStyle :
          styles.primarybtn}
      onPress={onPress}
      disabled={disabled}>

      <Text style={styles.text}>{text}</Text>
    </TouchableOpacity>
  );
}

export default PrimaryButton;

const styles = StyleSheet.create({
  primarybtn: {
    // fontFamily: FONT_NAME.Brown,

    borderRadius: widthPercentageToDP(4),
    backgroundColor: COLORS.ThemeColor,
    width: width * 0.90,
    height: height * 0.06,
    justifyContent: 'center',
    alignItems: 'center',
    alignSelf: 'center',
  },
  disabledStyle: {
    // fontFamily: FONT_NAME.Brown,
    backgroundColor: COLORS.LightBlue,
    width: width * 0.82,
    height: height * 0.07,
    justifyContent: 'center',
    alignItems: 'center',
    alignSelf: 'center',
  },
  text: {
    color: COLORS.White,
    fontWeight: 'bold',
    // fontStyle:''
    fontSize: FONT_SIZE.Medium,
    // fontFamily: FONT_NAME.Brown,
  },
});
