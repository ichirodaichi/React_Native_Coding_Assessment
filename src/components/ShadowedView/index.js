import React, { Children } from "react";
import { View, Text, TouchableOpacity, Image } from "react-native";
import { widthPercentageToDP } from "react-native-responsive-screen";
import { COLORS } from "../../utils/Colors";
import { FONT_SIZE } from "../../utils/FontSizes";
import IMAGE_PATHS from "../../utils/ImagePaths";


const ShadowedView = ({ extraStyle, children }) => {

    return (
        <View
            style={[{
                shadowOffset: {
                    width: 4,
                    height: 4,
                },
                shadowOpacity: 0.2,
                shadowRadius: 1.5,
                shadowColor: 'black',
                shadowRadius: 2,
                elevation: 5,
                padding: widthPercentageToDP(2),
                borderRadius: 14,
                flexDirection: 'row'
            }, extraStyle]}>
            {children}


        </View>
    )

}

export const ShadowedChild = ({ icon, firstText, secondText, isReversed }) => {
    return (

        <View style={{
            flex: 0.4,
            alignItems: 'center',
            flexDirection: isReversed? 'row-reverse':'row'
        }}>

            {icon &&
                <Image source={icon} style={{ width: 40 }}></Image>}
            <View style={{
                justifyContent: 'center',
                paddingHorizontal: 10,

            }} >

                <Text style={{
                    fontSize: FONT_SIZE.Small,
                    fontWeight: "400",
                    color: COLORS.Grey
                }}>
                    {firstText}
                </Text>
                <Text style={{
                    fontSize: FONT_SIZE.Small,
                    fontWeight: "400",
                    marginTop: 2,
                    color: COLORS.Black
                }}>
                    {secondText}
                </Text>
            </View>
        </View>
    )

}

export const ShadowedCenter = ({ isVisible = true, icon, isTouchable, onPress }) => {
    return (
        <TouchableOpacity
            disabled={!isTouchable}
            onPress={onPress}
            style={{ flex: 0.2, alignItems: 'flex-end' }}>
            {isVisible && <Image style={{ height: '100%' }} resizeMode={"contain"} source={icon} />
            }
        </TouchableOpacity>
    );

}

export const ShadowedText = ({ childText }) => {
    return (
        <View style={{ margin: 10, justifyContent: 'center' }}>
            <Text>{childText}</Text>
        </View>
    )
}


export default ShadowedView;

