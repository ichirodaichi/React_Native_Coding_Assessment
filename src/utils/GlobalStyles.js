import { StyleSheet } from 'react-native';
import { FONT_SIZE } from './FontSizes';
import { COLORS } from './Colors';
import { heightPercentageToDP, widthPercentageToDP } from 'react-native-responsive-screen';

export const GlobalStyles = StyleSheet.create({
  SliderHeaderText: {
    fontSize: FONT_SIZE.ExtraLarge,
    // fontFamily: FONT_NAME.Brown,
    color: COLORS.Black,
  },
  InputField: {
    fontSize: FONT_SIZE.Medium,
    color: COLORS.LightGrey2, //'#B2B2B2',
    height: 50,
  },
  baseview: {
    flex: 1,
    backgroundColor: COLORS.bgColor,
  },
  container: {
    flex: 0.9,
  },
  txt: {
    fontSize: FONT_SIZE.Small,
    fontWeight: 'bold',
    marginBottom: 4
  },
  inputText: {
    color: COLORS.ThemeColor,
    // marginTop: heightPercentageToDP(1),
    fontSize: FONT_SIZE.Large,
    fontWeight: 'bold'
  },
  inputContainer: {
    marginTop: heightPercentageToDP(2),
    alignItems: 'center'
  },
  container1: {
    fontSize: FONT_SIZE.Small,
    margin: 10,
    fontWeight: 'bold'
  },
  extraView: {
    width: '100%',
    height: heightPercentageToDP(8),
    backgroundColor: COLORS.White,
    flexDirection: 'row',
    justifyContent: 'space-between',

  },
  calButtonView: {
    backgroundColor: COLORS.White,
    height: 38,
    borderRadius: 4,
    shadowOffset: {
      width: 3,
      height: 3,
    },
    shadowOpacity: 0.1,
    shadowColor: COLORS.Grey,
    shadowRadius: 2,
    elevation: 4,
    alignItems: 'center',
    width: widthPercentageToDP(30)
  },
  calButtonText: {
    fontSize: FONT_SIZE.Medium,
    marginTop: 5,
    fontWeight: '600'
  },
  primaryView: {
    flex: 0.2,
    justifyContent: 'space-evenly',
    alignItems: 'center'
  },
  underlineStyle: {
    height: 1,
    width: '100%',
    backgroundColor: COLORS.LightBlue
  },
  exchangeTabView: {
    padding: 2,
    height: heightPercentageToDP(3.5),
    backgroundColor: COLORS.tabBg, 
    borderRadius: 6, 
    flexDirection: 'row',
    justifyContent: 'space-evenly', 
    alignItems: 'stretch'
  },
  buySellViewStyle:{
    justifyContent: 'center',
     flex: 1, 
     alignItems: 'center'
  },
  buySellText:{
    fontSize:FONT_SIZE.Small,
    color:COLORS.Grey
  }
});
