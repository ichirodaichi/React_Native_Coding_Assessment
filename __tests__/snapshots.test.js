import { render } from '@testing-library/react-native';
import React from 'react';
import { heightPercentageToDP } from 'react-native-responsive-screen';
import Header from '../src/components/Header';
import PrimaryButton from '../src/components/PrimaryButton';
import ShadowedView, { ShadowedChild } from '../src/components/ShadowedView';
import { COLORS } from '../src/utils/Colors';

test('Header renders correctly', () => {
    const snap = render(
        <Header
            headerText={"Hello"}
        />).toJSON();
    expect(snap).toMatchSnapshot();
});

test('Button renders correctly', () => {
    const snap = render(
        <PrimaryButton
            text={"Hello"}
            onPress={() => console.log("Working")} />
    ).toJSON();
    expect(snap).toMatchSnapshot();
});

test('Shadowed Component renders correctly', () => {
    const snap = render(
        <ShadowedView
            extraStyle={{
                width: '100%',
                height: heightPercentageToDP(8),
                backgroundColor: COLORS.White
            }} />
    ).toJSON();
    expect(snap).toMatchSnapshot();
});

test('Shadowed Child renders correctly', () => {
    const snap = render(
        <ShadowedChild
            firstText={"Bitcoin BTC"}
            secondText={"0.040141"} />
    ).toJSON();
    expect(snap).toMatchSnapshot();
});