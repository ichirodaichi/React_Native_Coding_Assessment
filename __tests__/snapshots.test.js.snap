// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#2F66F6",
      "borderRadius": undefined,
      "height": 93.38000000000001,
      "justifyContent": "center",
      "opacity": 1,
      "width": 675,
    }
  }
>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontSize": 5,
        "fontWeight": "bold",
      }
    }
  >
    Hello
  </Text>
</View>
`;

exports[`Header renders correctly 1`] = `
<View
  style={
    {
      "paddingBottom": undefined,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#D7D9E4",
        "elevation": 5,
        "flexDirection": "row",
        "height": NaN,
        "justifyContent": "center",
        "paddingHorizontal": undefined,
        "position": "relative",
        "shadowOffset": {
          "height": 1.5,
          "width": 0,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 1.5,
        "width": undefined,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": undefined,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Image
          resizeMode="contain"
          source="Back.png"
        />
      </View>
      <Text
        style={
          {
            "fontSize": 6,
            "fontWeight": "bold",
          }
        }
      >
        Hello
      </Text>
      <View
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Image
          resizeMode="contain"
          source="Info_light.png"
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`Shadowed Child renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 0.4,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      {
        "justifyContent": "center",
        "paddingHorizontal": 10,
      }
    }
  >
    <Text
      style={
        {
          "color": "#696F8C",
          "fontSize": 3.5,
          "fontWeight": "400",
        }
      }
    >
      Bitcoin BTC
    </Text>
    <Text
      style={
        {
          "color": "black",
          "fontSize": 3.5,
          "fontWeight": "400",
        }
      }
    >
      0.040141
    </Text>
  </View>
</View>
`;

exports[`Shadowed Component renders correctly 1`] = `
<View
  style={
    [
      {
        "borderRadius": 14,
        "elevation": 5,
        "flexDirection": "row",
        "padding": undefined,
        "shadowColor": "black",
        "shadowOffset": {
          "height": 4,
          "width": 4,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 2,
      },
      {
        "backgroundColor": "#FFFFFF",
        "height": undefined,
        "width": "100%",
      },
    ]
  }
/>
`;